core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -396,9 +395,23 @@ abstract class AbstractControllerBrokerRequestBatch(config: KafkaConfig,
   def addStopReplicaRequestForBrokers(brokerIds: Seq[Int],
                                       topicPartition: TopicPartition,
                                       deletePartition: Boolean): Unit = {
+    // A sentinel (-2) is used as an epoch if the topic is queued for deletion or
+    // does not have a leader yet. This sentinel overrides any existing epoch.
```

Comment:
The first part of this definitely makes sense, but what is the motivation for the second part? Why not use `LeaderAndIsr.NoEpoch`? Though I can't really think of what would cause this case to be hit.

Reply:
Indeed, `LeaderAndIsr.NoEpoch` makes more sense as a default value here. I did a mistake here. You're right. I don't think this should ever happen but I went on the defensive path with a default value in case.

Reply:
I have updated the PR.